### PR TITLE
Minimal support for Echo and Request-tag [RFC 9175]

### PR DESCRIPTION
**Pull request by szysas**
* Partial support for [RFC 9175](https://datatracker.ietf.org/doc/html/rfc9175)
  - Request-Tag support for blockwise transfers
  - Client side handling of echo challenge

### CODE REVIEW

**Comment by sbernard31** [comment]
I understand that you implement,

**For `echo` Option:**   
At client side, if server send `4.01 Unauthorized + Echo Option` resend request with `Echo` option.


**For `Request-tag` Option:**   
At client side, for block1(only?) you automatically add `request-tag` option.
At server side, if client send concurrent block request, you only keep the more recently started transfer (like before) and  now you are able to reject old block request (identified by `request-tag`) with `408 REQUEST_ENTITY_INCOMPLETE`.

Right ?

File: coap-core/src/main/java/com/mbed/coap/server/block/BlockRequestId.java [hunk 1]
```diff
@@ -24,7 +25,11 @@ class BlockRequestId {
     private final String uriPath;
     private final InetSocketAddress sourceAddress;
 
-    public BlockRequestId(String uriPath, InetSocketAddress sourceAddress) {
+    static BlockRequestId from(CoapRequest request) {
+        return new BlockRequestId(request.options().getUriPath(), request.getPeerAddress());
+    }
```
Comment (sbernard31) [inline coap-core/src/main/java/com/mbed/coap/server/block/BlockRequestId.java:30]: If you wanted to support concurrent block transfer you could add request-tag as part of the id right ?

Comment (szysas) [reply coap-core/src/main/java/com/mbed/coap/server/block/BlockRequestId.java:30]: Exactly

**Comment by szysas** [comment]
> I understand that you implement,
> 
> **For `echo` Option:** At client side, if server send `4.01 Unauthorized + Echo Option` resend request with `Echo` option.
> 
> **For `Request-tag` Option:** At client side, for block1(only?) you automatically add `request-tag` option. At server side, if client send concurrent block request, you only keep the more recently started transfer (like before) and now you are able to reject old block request (identified by `request-tag`) with `408 REQUEST_ENTITY_INCOMPLETE`.
> 
> Right ?

Exactly.

> So I don't know if you want to add request-tag automatically with coap+tcp ?

With current change, it is not used in coap+tcp as that has a different builder class (`CoapServerBuilderForTcp`)

**Review by sbernard31 — APPROVED** [review]
A good first step to me ! :+1:

**Pull request merged**